source/common/http/http2/codec_impl.cc

```diff
@@ -194,6 +215,15 @@ void ConnectionImpl::StreamImpl::submitTrailers(const HeaderMap& trailers) {
   ASSERT(rc == 0);
 }
 
+void ConnectionImpl::StreamImpl::submitMetadata() {
+  ASSERT(stream_id_ > -1);
```

Comment:
We should never encode metadata on stream 0/connection either, right? I think this can be > 0?